### PR TITLE
PXC 5.7.40 merge part1

### PR DESCRIPTION
**Pull request by VarunNagaraju**

### CODE REVIEW

**Comment by VarunNagaraju** [comment]
Jenkins test results of the part 1 merge: https://pxc.cd.percona.com/view/PXC%205.7/job/pxc-5.7-param/359/#showFailuresLink
Two test failures: main.merge and main.merge_mmap.

File: MYSQL_VERSION [hunk 1]
```diff
@@ -1,13 +1,5 @@
...
-MYSQL_VERSION_EXTRA=-42
-PXC_VERSION_EXTRA=61
-||||||| b0a7dc2da2e
-MYSQL_VERSION_PATCH=39
-MYSQL_VERSION_EXTRA=-42
-=======
 MYSQL_VERSION_PATCH=40
 MYSQL_VERSION_EXTRA=-43
->>>>>>> fde2cca41b14185d354d2f1ce94dc3ac5b3cbd39
+PXC_VERSION_EXTRA=62
```
Comment (kamil-holubicki) [inline MYSQL_VERSION]: This should be bumped to next even number, so 63

File: packaging/WiX/mysql_server.wxs.in [hunk 2]
```diff
@@ -76,44 +76,6 @@
...
-              Key="SOFTWARE\Microsoft\VisualStudio\12.0\VC\Runtimes\x86"
-              Name="Version"
-              Type="raw" />
-        </Property>
-        <Condition Message="This application requires Visual Studio 2013 x86 Redistributable. Please install the Redistributable then run this installer again.">
-          Installed OR VS12REDISTX86
-        </Condition>
-      <?endif?>
-=======
       <?if "@Platform@"="x64"?>
```
Comment (kamil-holubicki) [inline packaging/WiX/mysql_server.wxs.in:79]: Interesting, that we had a conflict in this file... :)

File: sql/log.h [hunk 3]
```diff
@@ -1004,15 +1004,12 @@ bool log_syslog_find_facility(char *f, SYSLOG_FACILITY *rsf);
...
-<<<<<<< HEAD
 #ifdef WITH_WSREP
 IO_CACHE* wsrep_get_trans_log(THD * thd, bool transaction);
 bool wsrep_trans_cache_is_empty(THD *thd);
 void wsrep_thd_binlog_flush_pending_rows_event(THD *thd, bool stmt_end);
 void wsrep_thd_binlog_trx_reset(THD * thd);
 #endif /* WITH_WSREP */
-||||||| b0a7dc2da2e
-=======
 /* 26 for regular timestamp, plus 7 (".123456") when using micro-seconds */
```
Comment (kamil-holubicki) [inline sql/log.h:1007]: My suggestion is to move the new block before #ifdef WITH_WSREP.
This way we will have WSREP-related things at the end of the file, not in the middle

File: build-ps/percona-server.spec [hunk 4]
```diff
@@ -91,8 +91,8 @@
 
 # Version for compat libs
 %if 0%{?rhel} > 6
-%global compatver             5.6.28
-%global percona_compatver     76.1
+%global compatver             5.6.51
```
Comment (kamil-holubicki) [inline build-ps/percona-server.spec]: content of build-ps directory should not be changed

**Review by kamil-holubicki — CHANGES_REQUESTED** [review]

File: build-ps/percona-server.spec (hunk 4 above)
Comment (VarunNagaraju) [reply build-ps/percona-server.spec]: Done.

File: sql/log.h (hunk 3 above)
Comment (VarunNagaraju) [reply sql/log.h:1007]: Done.

**Review by kamil-holubicki — APPROVED** [review]
lgtm

**Comment by VarunNagaraju** [comment]
Jenkins run with PXB 2.4.27: https://pxc.cd.percona.com/job/pxc-5.7-param/363/
Test failures: 
rpl.rpl_mts_gtids_restart_slave_io_lost_trx_with_master_delay : Seems like a flaky test. Verified.

**Pull request merged**